### ISSUE TITLE
Incorrect traction battery voltage

### ISSUE THREAD

**Issue by Memosis** [flobz/psa_car_controller#765]
Battery voltage sensor seems to be providing invalid data for traction battery. The voltage is way lower than what is shown via OBD (e.g. now at 82% SoC OBD says 426V, this sensor shows 340V in HA) and also the voltage drops or increases irrespectively of traction battery percentage. E.g. when I charged the traction battery from 73% to 82%, voltage shown DROPPED from 346V to 340V.

I was looking for this issue here on Github and it seems this might be it? The sensor shows data from accessory 12V battery (just multiplied by some number) and not from real traction battery?

https://github.com/flobz/psa_car_controller/discussions/354

![image](https://github.com/flobz/psa_car_controller/assets/6164309/58e895d5-cc45-43a2-902c-bd7419e1d1ad)
![image](https://github.com/flobz/psa_car_controller/assets/6164309/52abd6db-0b00-44a5-ae9c-eaed2a73706b)
![image](https://github.com/flobz/psa_car_controller/assets/6164309/c517919b-ff75-4d7c-a605-79652d3886fa)

**Comment by Memosis** [comment]
My 12V battery is currently at 11.3 volts so that would mean multiplication by ~30 to get those 340V reported?
![image](https://github.com/flobz/psa_car_controller/assets/6164309/e6562735-8ff1-4c80-ad66-97357aef4500)

**Comment by p4block** [comment]
This would explain lots of things. I can also replicate that the value for the battery voltage has no resemblance at all to the real voltage output by OBD

**Comment by tomechio** [comment]
For me this value has been percentage of the 12V battery state of charge.
Meaning that your 340V (85*4) would be 85% SOC of the 12V battery.
And this value is very useful. Few times I have noticed that for some reason the car is awake all the time and it consumes the 12V battery a lot.

**Comment by Memosis** [comment]
You are right, it's 0-100% multiplied by 4. Although it's useful to (also) have voltage of 12V battery, is there a value for traction battery?
![image](https://github.com/flobz/psa_car_controller/assets/6164309/497783bd-7ece-4ccb-b1a8-a90f9789087a)

**Comment by Memosis** [comment]
So I've renamed label for this sensor and put multiplication there (based on my OBD reading) so it shows correctly for 12V battery
![image](https://github.com/flobz/psa_car_controller/assets/6164309/909e080d-5f83-4496-b146-f08633bd678c)
![image](https://github.com/flobz/psa_car_controller/assets/6164309/b5836c2f-ea03-4156-bd2b-265b1dd46153)

**Comment by papageigit** [comment]
What does the value return when the car is on and in ready? Do you get ~14V via API? Does your API returned voltage match the voltage measured by your OBD device?

**Comment by Memosis** [comment]
That's a great question. You are right that when you turn the car on the OBD 12V shows something like 13.7V (as DC to DC convertor provides the voltage directly from the traction battery (and hopefully also slowly recharges the 12V battery from time to time)). 

When I check the voltage history in my HA for the last week, it shows minimum 318 (79.5%) and maximum 348 (87%) with no sudden spikes. So it seems this reported voltage is always from the 12V battery only, ignoring OBD voltage.

![image](https://github.com/flobz/psa_car_controller/assets/6164309/932935bf-f5a2-49dd-b1c3-7fd35f9e9e95)

**Comment by papageigit** [comment]
If more people can confirm that this is actually the SoC of the LV battery, please consider creating a pull request over at https://github.com/Flodu31/HomeAssistant-PeugeotIntegration to get this changed.

Perhaps maybe someone can update the template to the modern format rather than the legacy format as well? I would do this myself but my PSACC is not functioning so I can't test.

**Comment by gerar19** [comment]
It's a % soh of 12V Battery.

**Comment by HenningS81** [comment]
Yes, they did that update at a service center. We got our e208 May last year and the update was done 1st December, so fairly recent.

**Comment by gernot-h** [comment]
Can someone probably translate this from HA terms to pure psacc values for me? 

We also had a discussion on the `battery` field in the `get_vehicleinfo` answer over at https://github.com/flobz/psa_car_controller/issues/755#issuecomment-1940596470 - is this the value you're referring to here?

**Comment by gernot-h** [comment]
Perhaps you even want to have a look at #755 where I raised the question whether usage of `-R` (force data refresh by every X mins) should probably be limited or at least "safe" values (> 15 mins?) should be recommended in the help.

**Comment by HenningS81** [comment]
I'm certain that the battery:voltage value is the SoC of the 12 V battery in %. I've read somewhere that it is the same in fuel driven Stellantis cars.
I wouldn't recommend wake up calls more often than once per hour. For my car I can say that every call leads to 0.5% drop in the SoC of the 12 V battery. But you wouldn't gain much advantage of waking the car more often aynway, maybe a higher accuracy for the temperature. Because whenever the car is active, may it be charging or driving, it sends data more frequently anyway. In case of driving exactly every two minutes.

**Comment by HenningS81** [comment]
![image](https://github.com/flobz/psa_car_controller/assets/43236351/5d21dc3b-8580-444b-bdab-ea399254e929)
The graph shows the SoC of the traction battery, so this was a short drive and a charge afterwards. The colorful bar at the top shows every timestamp transmitted by Stellantis (to be exact: ["energy"][0]["updated_at"])

**Comment by HenningS81** [comment]
![image](https://github.com/flobz/psa_car_controller/assets/43236351/28c87647-c843-4451-9e6a-dbe02fb3b9f7)
That's how the 12 V battery get's drained by the wakeup calls.
And because I didn't believe that my garage checked the battery and measured for leak currents. Without any findings.

**Comment by gernot-h** [comment]
By the way, don't know exactly how it's for BEVs, but in my Astra PHEV, there are two 12V batteries, one in the motor compartment, one much smaller one in the centre console below the armrest. I couldn't find reliable information about their work split yet, but I guess we're talking about the (larger) one in the motor compartment, which seems to be needed for starting the car (in contrast to the smaller one probably driving radio & friends), right?

**Comment by sod75** [comment]
My Peugeot PHEV also has 2x12v batteries. I only know because I had an issue and the garage told me they had to replace "One of the two 12V batteries" when asking for the root cause...

**Comment by gernot-h** [comment]
> My Peugeot PHEV also has 2x12v batteries. I only know because I had an issue and the garage told me they had to replace "One of the two 12V batteries" when asking for the root cause...

Yes. Opel Astra L and Peugeot 308 are nearly identical "under the hood", its common best practice for Astra owners to also  read the Peugeot manual as it covers some details in a much better way, e.g. the fact with the two 12V batteries. So RTFM ;-) 

And you should really read the manual here as you risk breaking some electronic parts if you (or someone in the garage...) unconnect the 12 V batteries in the wrong order (IIRC, the small one in the centre console always need to be unconnected first before unconnecting the "main" battery in the motor compartment).

Sorry for off-topic...

**Comment by Jordan87** [comment]
Hi

For me indeed "Battery voltage" seems to be 12V main battery charge level. with Range 0-400 (0,25% resolution) where 400 is 100%, 0 is most probably 0% :)

Yestarday I was doing some test with disharging my 12V battery to simulate and check how car will behave if I i.e. connect my DVR  Camera in parking mode with dedicated module.
When I started "Battery voltage" entity was showing value 380V which i think was 95% (real voltage on battery was 13,14V)
My main electronic load was set to stop dishachring around 11.8V, with this level "Battery voltage" was showing 200V which I think was 50%

![image](https://github.com/flobz/psa_car_controller/assets/10414918/e38412af-5133-4d42-bf40-9cf2f5569301)

![image](https://github.com/flobz/psa_car_controller/assets/10414918/8c8125e0-a09f-4a7c-92a5-c67c45f0082e)

On screen from @HenningS81 we see that when battery was "dead" it showed even 0V

**Comment by gerar19** [comment]
@Jordan87 You do not take into account that the value you say of 400V is just a manually created sensor that is set to multiply the % of voltage that the API returns by 400.
So you don't have to take it into account.
The % of the voltage is from the 12V battery, you don't need to give it any more thought.
If you want the correct Vs, change this multiplication in your sensor in the corresponding yaml.

**Comment by Jordan87** [comment]
Ohh, you are correct, now I see that it is not directly from psacc but was created base on https://github.com/Flodu31/HomeAssistant-PeugeotIntegration
I should correct this one.
`      308_battery_voltage:
        friendly_name: "Battery Voltage"
        unit_of_measurement: "V"
        value_template: "{{ state_attr('sensor.peugeot_308', 'battery')['voltage'] * 4 }}"
        icon_template: mdi:flash-triangle-outline`

Thanks

**Comment by Memosis** [comment]
Yes, as I was showing previously. ;) 
![image](https://github.com/flobz/psa_car_controller/assets/6164309/a3b1e658-cb7c-42c7-90fb-d47a7181649d)

I will also perform new measurement with OBD and multimeter at the same time to finetune the multiplcation value. I previously had smaller 60Ah EFB battery, now I have 72Ah AGM (which should have been configured in BMS by the service center, though the reported voltage should be the same I believe)

**Comment by Jordan87** [comment]
Im sure its not about volts but % level of charge 12V battery. 400V was in my case ~13.2V but 200V was around 11,8V
With volts this cant be 50% drop.

**Comment by Memosis** [comment]
When my 12V battery was just 11.1V, the reported value was ~83% ("332V" in the incorrect calculation). At 11.3V it was ~85% ("340V"). This is Citroen e-C4.

**Comment by Jordan87** [comment]
Here you can see my graphs even with exact spots (dips in voltage) when car was woke up. 200V= 11.8V
![image](https://github.com/flobz/psa_car_controller/assets/10414918/4dfd98e5-5044-40c2-ad7c-dd3396e99293)

**Comment by Memosis** [comment]
I finally managed to measure my 12V battery in e-C4 using physical multimeter. See below my process. Notes:
1) OBD/Car Scanner reports the voltage lower by 0.9V. Friend of mine who has also e-C4 was showing me his Car Scanner values previously and they were higher, so this might be just an issue with my OBD adapter/Car Scanner profile.
2) I assume the reported value to Stellantis cloud is the last one measurement (with traction battery disconnected). Battery was 12.77V at that moment and cloud reported value of 93.
Let me know your thoughts...

![image](https://github.com/flobz/psa_car_controller/assets/6164309/6e60a039-fa88-4406-b120-1716e08c1ae8)

**Comment by papageigit** [comment]
I believe the voltage reported in Car Scanner is the voltaged measured at the OBD port by the OBD tool. Cheap OBD tools could provide a reading that isn't 100% correct due to cheap measuring components.

**Comment by Memosis** [comment]
> You have low battery level in your multimeter

That battery icon, right. I thought it was just assuring me I measure DC (battery). :D I will replace it, thanks, though the measured 12V values seem plausible, I don't expect no significant difference. What surprised me was that OBD. If I get a chance to meet with my friend, we should try both OBD adapters/car scanners.

**Comment by Memosis** [comment]
> I believe the voltage reported in Car Scanner is the voltaged measured at the OBD port by the OBD tool. Cheap OBD tools could provide a reading that isn't 100% correct due to cheap measuring components.

I see. My OBD is iCar Pro. Friend of mine has much cheaper type, just for Android, go figure. :D 
https://www.amazon.de/dp/B07SQLY39V/ref=pe_27091401_487024491_TE_item
